### PR TITLE
[GPT-747][MAJOR] Unified compiler

### PR DESCRIPTION
**Pull request by nickbalestra**

### CODE REVIEW

File: packages/oc-template-jade-compiler/to-be-published/hash-builder.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
```
Comment (matteofigus) [inline packages/oc-template-jade-compiler/to-be-published/hash-builder.js]: We may want to investigate https://github.com/puleos/object-hash
It is widely used, dep free, well maintained, apparently, and defaults to sha1+hex.

Comment (matteofigus) [inline packages/oc-template-jade-compiler/to-be-published/hash-builder.js]: I can take care of it inside OC as separate PR

Comment (nickbalestra) [inline packages/oc-template-jade-compiler/to-be-published/hash-builder.js]: Nice! We can definitely use that instead. Thanks for pointing that out 💅

**Comment by nickbalestra** [comment]
addressed #17 ✅

**Comment by nickbalestra** [comment]
Addresses #18

**Comment by nickbalestra** [comment]
addresses #20

**Comment by nickbalestra** [comment]
```bash
Test Suites: 16 passed, 16 total
Tests:       43 passed, 43 total
Snapshots:   30 passed, 30 total
Time:        6.374s
Ran all test suites.
-----------------------------------------|----------|----------|----------|----------|----------------|
File                                     |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------------------------------------|----------|----------|----------|----------|----------------|
All files                                |    92.34 |    72.22 |     97.5 |    92.27 |                |
 infinite-loop-loader                    |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
 infinite-loop-loader/lib                |    96.55 |    82.61 |      100 |    96.55 |                |
  infinite-loop-loader.js                |       90 |    63.64 |      100 |       90 |              9 |
  transformLoopWithLimit.js              |      100 |      100 |      100 |      100 |                |
 oc-external-dependencies-handler        |    94.12 |       70 |      100 |    94.12 |                |
  index.js                               |    94.12 |       70 |      100 |    94.12 |             35 |
 oc-get-unix-utc-timestamp               |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
 oc-hash-builder                         |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
 oc-minify-file                          |    83.33 |       25 |      100 |    83.33 |                |
  index.js                               |    83.33 |       25 |      100 |    83.33 |          24,25 |
 oc-template-jade                        |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
 oc-template-jade-compiler               |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
 oc-template-jade-compiler/lib           |    94.74 |       75 |      100 |    94.69 |                |
  compile.js                             |    91.89 |       60 |      100 |    91.67 |       31,47,65 |
  compileServer.js                       |    94.44 |     62.5 |      100 |    94.44 |             27 |
  compileStatics.js                      |      100 |    94.44 |      100 |      100 |             14 |
  compileView.js                         |    90.91 |       50 |      100 |    90.91 |          19,48 |
 oc-template-jade-compiler/lib/resources |      100 |      100 |      100 |      100 |                |
  strings.js                             |      100 |      100 |      100 |      100 |                |
 oc-template-jade/lib                    |    76.92 |      100 |    66.67 |       75 |                |
  getCompiledTemplate.js                 |       50 |      100 |        0 |       50 |          5,6,7 |
  getInfo.js                             |      100 |      100 |      100 |      100 |                |
  render.js                              |      100 |      100 |      100 |      100 |                |
 oc-view-wrapper                         |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
 oc-webpack                              |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
 oc-webpack/lib/compiler                 |    76.19 |    58.33 |      100 |    76.19 |                |
  index.js                               |    76.19 |    58.33 |      100 |    76.19 | 19,25,26,30,36 |
 oc-webpack/lib/configurator             |      100 |      100 |      100 |      100 |                |
  index.js                               |      100 |      100 |      100 |      100 |                |
-----------------------------------------|----------|----------|----------|----------|----------------|

```

**Comment by nickbalestra** [comment]
@matteofigus should be ready to be reviewd/merged.
note: I'll bring the handlebars template as a follow-up PR as this is already pretty big

**Comment by mattiaerre** [comment]
let's aim for 100% code coverage @nickbalestra 😊

File: .travis.yml [hunk 2]
```diff
@@ -2,7 +2,8 @@ language: node_js
 
 matrix:
   include:
-    - node_js: "4"
-    - node_js: "5"
+    # - node_js: "4"
+    # - node_js: "5"
```
Comment (matteofigus) [inline .travis.yml]: Can we just remove 4 and 5? Thanks

File: packages/infinite-loop-loader/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# infinite-loop-loader
+[![Build Status](https://travis-ci.org/nickbalestra/infinite-loop-loader.svg?branch=master)](https://travis-ci.org/nickbalestra/infinite-loop-loader)
```
Comment (matteofigus) [inline packages/infinite-loop-loader/README.md]: wrong travis url

File: packages/infinite-loop-loader/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+[![Standard - JavaScript Style Guide](https://img.shields.io/badge/code%20style-standard-brightgreen.svg)](http://standardjs.com/)
```
Comment (matteofigus) [inline packages/infinite-loop-loader/README.md]: remove standard badge

File: packages/oc-external-dependencies-handler/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+# oc-external-dependencies-handler
```
Comment (matteofigus) [inline packages/oc-external-dependencies-handler/README.md]: Can we add just a little description of what this is. Same for all the other modules. At least one single sentence that matches package's "description" field.

File: packages/oc-get-unix-utc-timestamp/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# oc-get-unix-utc-timestamp
```
Comment (matteofigus) [inline packages/oc-get-unix-utc-timestamp/README.md]: Again, just a little description. Also, can we remove 'oc' from the name? I think this is more of a generic "get unix utc timestamp" module.

File: packages/oc-get-unix-utc-timestamp/test/oc-get-unix-utc-timestamp.test.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+const getTimeStamp = require('../index.js');
+
+test('Generates correct timestamps everytime its invoked', () => {
```
Comment (matteofigus) [inline packages/oc-get-unix-utc-timestamp/test/oc-get-unix-utc-timestamp.test.js]: it's

File: packages/oc-template-jade-compiler/lib/compile.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+const async = require('async');
+const _ = require('lodash');
+const fs = require('fs-extra');
+const path = require('path');
+const compileView = require('./compileView');
+const compileServer = require('./compileServer');
+const compileStatics = require('./compileStatics');
+const getUnixUtcTimestamp = require('oc-get-unix-utc-timestamp');
+
+// OPTIONS
```
Comment (matteofigus) [inline packages/oc-template-jade-compiler/lib/compile.js]: can we make this list alphabetic. Also, I don't see here "ocPackage"

File: packages/infinite-loop-loader/README.md (hunk 3 above)
Comment (nickbalestra) [inline packages/infinite-loop-loader/README.md]: Good catch! This could be removed, as we have now the travis badge on root for the whole monorepo

**Comment by nickbalestra** [comment]
@matteofigus done

**Pull request merged**